### PR TITLE
DDF-4160 Update Basic Search Form to use configurable attribute for type

### PR DESCRIPTION
**Pull request by andrewzimmer**
#### What does this PR do?
Update Basic Search Form to use configurable attribute for type; right now it is hardcoded to use the datatype attribute.
#### Who is reviewing it? 
@bdeining @Bdthomson @alanchampion 

#### Select relevant component teams: 
https://github.com/orgs/codice/teams
#### Ask 2 committers to review/merge the PR and tag them here.

@andrewkfiedler
@djblue
#### How should this be tested?

- build/run
- Run a Basic Search with a specific Match Type selected 
- verify type selected was searched with "datatype" and "metadata-content-type" 
screen shot 2018-09-24 at 12 17 30 pm

- in Admin search ui config set Basic Search Match Type Metacard Attribute to a valid metacard attribute
screen shot 2018-09-24 at 12 09 42 pm

- run another search and verify type selected was searched with "metadata-content-type" and the attribute set in admin config
- in Admin search ui config set Basic Search Match Type Metacard Attribute to an invalid metacard attribute
- run another search and verify type selected was searched with "metadata-content-type" and "datatype"
#### What are the relevant tickets?
[DDF-4160](https://codice.atlassian.net/browse/DDF-4160)
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: ui/packages/catalog-ui-search/src/main/webapp/component/query-basic/query-basic.view.js [hunk 1]
```diff
@@ -60,6 +62,12 @@ define([
     return nested
   }
 
+  function getMatchTypeAttribute() {
+    return metacardDefinitions.metacardTypes[properties.basicSearchMatchType]
+      ? properties.basicSearchMatchType
+      : 'datatype'
```
Comment (Bdthomson) [inline ui/packages/catalog-ui-search/src/main/webapp/component/query-basic/query-basic.view.js]: Do we actually want to be defaulting here if the property is not set? I would think setting the default value in `org.codice.ddf.catalog.ui.config` would be enough.

Comment (andrewzimmer) [reply ui/packages/catalog-ui-search/src/main/webapp/component/query-basic/query-basic.view.js]: This is more for the case of an invalid metacard attribute name being entered in the config. Choices were to either exclude from the search or fallback to `datatype`. me and @bdeining decided on the latter

**Comment by bdeining** [comment]
build now

**Comment by willwill96** [comment]
hero started

**Comment by Bdthomson** [comment]
build now

**Comment by LinkMJB** [comment]
Build will not start until the queue is caught up with. Please see https://github.com/jenkinsci/ghprb-plugin/issues/707 for more information

**Comment by willwill96** [comment]
🎉 Hero Successful 🎉 
1. Quickbuilt and installed
2. Verified in admin that the new configurable attribute was present
datatype admin
3. Verified that a search reflected the new attribute
datatype search
4. Changed the attribute to a valid attribute ("id") and verified that search reflected it.
different datatype search
5. Changed the attribute to an invalid attribute and verified that "datatype" was defaulted to
datatype invalid

6. Changed the attribute back to a valid attribute and verified that it still worked correctly.
7. Spot checking around UI for bugs

**Comment by bdeining** [comment]
build now

**Pull request merged**